File: demo/top_n_buggy.py
"""Deliberately buggy: off-by-one in the slice, used as the running demo for ProofKit.

Usage: python top_n_buggy.py <item> <item> ... <n>
The last argument is n; everything before it is the list of items.
"""
import sys


def top_n(items, n):
    sorted_items = sorted(items, reverse=True)
    return sorted_items[n]  # bug: should be sorted_items[:n]


if __name__ == "__main__":
    *item_args, n_arg = sys.argv[1:]
    items = [int(x) for x in item_args]
    print(top_n(items, int(n_arg)))

File: demo/top_n_buggy.js
// Deliberately buggy: off-by-one, same shape of bug as top_n_buggy.py — used
// to prove ProofKit is language-agnostic (it just wraps a shell command).
//
// Usage: node top_n_buggy.js <item> <item> ... <n>

function topN(items, n) {
  const sorted = [...items].sort((a, b) => b - a);
  return sorted[n]; // bug: should be sorted.slice(0, n)
}

const args = process.argv.slice(2).map(Number);
const n = args.pop();
const result = topN(args, n);
console.log(result.toFixed(2)); // throws when result is undefined (n out of bounds)
